codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/endpoints/discovery/EndpointDiscoveryIntegration.kt

```diff
@@ -50,10 +62,11 @@ class EndpointDiscoveryIntegration : KotlinIntegration {
     override fun customizeMiddleware(
         ctx: ProtocolGenerator.GenerationContext,
         resolved: List<ProtocolMiddleware>,
-    ): List<ProtocolMiddleware> = super.customizeMiddleware(ctx, resolved) + listOf(DiscoveredEndpointMiddleware)
+    ): List<ProtocolMiddleware> = super.customizeMiddleware(ctx, resolved) + listOf(DiscoveredEndpointErrorMiddleware)
 
-    override fun enabledForService(model: Model, settings: KotlinSettings): Boolean =
-        model.expectShape<ServiceShape>(settings.service).hasTrait<ClientEndpointDiscoveryTrait>()
+    override fun enabledForService(model: Model, settings: KotlinSettings): Boolean = isEnabledFor(model, settings)
```

Comment:
Instead of repeating the integration's `isEnabled` check, can this just be `true`?

Reply:
Sorry I got this mixed up thinking it was the middleware's `isEnabledFor` function, you can ignore this